### PR TITLE
Fix ordering of children of featured pages

### PR DESCRIPTION
**Pull request by guy-roberts**
### Context
Small fixes

### Changes proposed in this pull request
- The links in the block for 'Get help to improve your practice' are now ordered by position
- '(' and ')' characters can now be used in titles, but they will be ignored in the url slug
- The cms link to preview works correctly
- After editing a page, the user is now taken to the list, rather than the show page
- Change docker file to use env vars for postgres

### Guidance to review
Verify each of the above.
For the docker change, run docker-compose build, docker-compose up and check that the container has a postgres service

### CODE REVIEW

File: app/controllers/content_pages_controller.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create
   # PATCH/PUT /content_pages/1
   def update
     if @content_page.update(content_page_params)
-      redirect_to @content_page, notice: "Content page was successfully updated."
+      redirect_to content_pages_path(@content_page), notice: "Content page was successfully updated."
     else
```
Comment (guy-roberts) [inline app/controllers/content_pages_controller.rb:45]: After editing, go to the list of pages, nobody wants to see the Show page !

File: app/models/content_page.rb [hunk 2]
```diff
@@ -5,8 +5,10 @@ class ContentPage < ApplicationRecord
   scope :top_level, -> { where("parent_id IS NULL") }
   scope :order_by_position, -> { order("position ASC") }
 
-  ONLY_ALPHA_NUMERIC_COMMA_HYPHEN_AND_SPACE = /\A[a-zA-Z0-9,.\- ]+\Z/.freeze
-  validates :title, format: { with: ONLY_ALPHA_NUMERIC_COMMA_HYPHEN_AND_SPACE, message: "should only contain alphabetic, numeric, comma, fullstop, hyphen and space characters" }
+  CHARS_TO_OMIT_FROM_SLUG = ",.()".freeze
+  ONLY_ALPHA_NUMERIC_COMMA_HYPHEN_SPACE_AND_ROUND_BRACES = /\A[a-zA-Z0-9,.\-\(\) ]+\Z/.freeze
+  TITLE_FORMAT_ERROR_MESSAGE = "should only contain alphabetic, numeric and -#{CHARS_TO_OMIT_FROM_SLUG}".freeze
+  validates :title, format: { with: ONLY_ALPHA_NUMERIC_COMMA_HYPHEN_SPACE_AND_ROUND_BRACES, message: TITLE_FORMAT_ERROR_MESSAGE }
   validates :title, presence: true, uniqueness: true
```
Comment (guy-roberts) [inline app/models/content_page.rb:12]: Allow ( and ) in page titles

File: app/views/content_pages/_details_for_a_single_page.erb [hunk 3]
```diff
@@ -6,7 +6,7 @@
       <span class="links">
         <%= link_to 'Edit', edit_content_page_path(page_to_render), { :class => "govuk-link" } %>
         |<%= link_to 'Add a child page', new_content_page_path(:parent_id => page_to_render.id) %>
-        |<%= link_to 'Preview', '/content/' + page_to_render.slug, target: "_blank" %>
+        |<%= link_to 'Preview', path_for_this_page(page_to_render), target: "_blank" %>
       </span>
```
Comment (guy-roberts) [inline app/views/content_pages/_details_for_a_single_page.erb:10]: This is a bug fix, no ticket

File: docker-compose.yml [hunk 4]
```diff
@@ -25,6 +25,7 @@ services:
       - ./init.sql:/docker-entrypoint-initdb.d/init.sql
     ports:
       - "5432:5432"
+    env_file: .env
```
Comment (guy-roberts) [inline docker-compose.yml:29]: Docker was not loading env vars for Postgres. We still do not know why it worked without, but this fixes it

File: spec/models/content_page_spec.rb [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 RSpec.describe ContentPage, type: :model do
   let(:expected_exception_message) do
-    /should only contain alphabetic, numeric, comma, fullstop, hyphen and space characters/
+    /#{ContentPage::TITLE_FORMAT_ERROR_MESSAGE}/
   end
```
Comment (ebrett) [inline spec/models/content_page_spec.rb:6]: I generally prefer the test not rely on the implementation, so I would have just stayed with the explicit expectation.

**Review by ebrett — APPROVED** [review]
Needs to be rebased with `main`

**Review by marcrice12 — APPROVED** [review]
great

File: app/models/content_page.rb (hunk 2 above)
Comment (marcrice12) [reply app/models/content_page.rb:12]: we should really have tests for this validation

File: spec/models/content_page_spec.rb (hunk 5 above)
Comment (guy-roberts) [reply spec/models/content_page_spec.rb:6]: Got you

**Pull request merged**